examples/nodejs/lambda-examples/cloudwatch-metrics/infrastructure/lib/momento-metrics-stack.ts

```diff
@@ -0,0 +1,337 @@
+    });
+
+    logGroup.addMetricFilter('ExampleMetricFilterResponseSize', {
+      metricNamespace: 'MomentoMetricsCDKExample',
+      metricName: 'Response Size (bytes)',
+      filterPattern: FilterPattern.literal('{$.responseSize >= 0}'),
+      metricValue: '$.responseSize',
+      unit: Unit.BYTES,
+    });
+
+    logGroup.addMetricFilter('ExampleMetricFilterGrpcStatusCode', {
+      metricNamespace: 'MomentoMetricsCDKExample',
+      metricName: 'GRPC Status Code',
+      filterPattern: FilterPattern.literal('{ $.status >= 0 }'),
+      metricValue: '$.status',
+    });
+  }
+
+  addWidgetsToDashboard(dashboard: cdk.aws_cloudwatch.Dashboard) {
+    const latency = new GraphWidget({
+      left: [
+        new Metric({
+          metricName: 'Duration (Latency)',
+          namespace: 'MomentoMetricsCDKExample',
+        }),
+      ],
+      title: 'Duration (Latency)',
+    });
+
+    const errorCodes = new GraphWidget({
+      left: [
+        new Metric({
+          metricName: 'GRPC Status Code',
+          namespace: 'MomentoMetricsCDKExample',
+        }),
+      ],
+      title: 'GRPC Error Codes',
+      period: cdk.Duration.minutes(1),
+      leftYAxis: {
+        label: 'Error Code',
+        max: 16,
+      },
+      leftAnnotations: [
+        {
+          value: 4,
+          fill: Shading.ABOVE,
+          label: 'GRPC Error Codes >= 4',
+        },
+      ],
+    });
+
+    const messageSizes = new GraphWidget({
+      left: [
+        new Metric({
+          metricName: 'Response Size (bytes)',
+          namespace: 'MomentoMetricsCDKExample',
+          label: 'Response Size (bytes)',
+        }),
+        new Metric({
+          metricName: 'Request Size (bytes)',
+          namespace: 'MomentoMetricsCDKExample',
+          label: 'Request Size (bytes)',
+        }),
+      ],
+      title: 'Request and Response Sizes in Bytes',
+      period: cdk.Duration.minutes(1),
+      statistic: Stats.AVERAGE,
+    });
+
+    // Add the first three widgets in the first row of the dashboard
+    dashboard.addWidgets(latency, errorCodes, messageSizes);
+
+    const transactionsPerSecond = new SingleValueWidget({
+      metrics: [
+        new Metric({
+          metricName: 'Response Size (bytes)',
+          namespace: 'MomentoMetricsCDKExample',
+          label: 'Count of responses per second',
+          statistic: Stats.SAMPLE_COUNT,
+        }),
+      ],
+      period: cdk.Duration.seconds(1),
+      height: 6,
+      sparkline: true,
+      title: 'Transactions per second',
+    });
+
+    const numberOfRequests = new GraphWidget({
+      left: [
+        new Metric({
+          metricName: 'Request Size (bytes)',
+          namespace: 'MomentoMetricsCDKExample',
+          label: 'Number of requests per second',
+        }),
+      ],
+      title: 'Number of requests',
+      period: cdk.Duration.minutes(1),
+      statistic: Stats.SAMPLE_COUNT,
+    });
+
+    const totalBytesSentReceived = new GraphWidget({
+      left: [
```

Comment:
let's do p50 and p99 on this one. and if we are able to get the request type dimension added then I think we should do "get p50", "get p99", "set p50", "set p99".